src/safeds/data/tabular/containers/_table.py

```diff
@@ -1057,9 +1057,9 @@ def add_rows(self, rows: list[Row] | Table) -> Table:
         2  5  6
         """
         if isinstance(rows, Table):
-            if rows.number_of_rows == 0:
+            if rows.number_of_columns == 0 or rows.number_of_rows == 0:
```

Comment:
The second check is not really necessary since number_of_colums == 0 => number_of_rows == 0